File: src/offerings/location.py
class Location:
    def __init__(self, address: str, city: str, province: str):
        self.address = address
        self.city = city
        self.province = province
       
    def full_address(self) -> str:
        return f"{self.address}, {self.city}, {self.province}"

    def __str__(self) -> str:
        return self.full_address()

    def get_city(self) -> str:
        return self.city

    def set_city(self, city: str) -> None:
         self.city = city

    def get_province(self) -> str:
        return self.province

    def set_province(self, province: str) -> None:
        self.province = province

File: src/main/java/offerings/Location.java
package offerings;

public class Location {

    private String address;
    private String city;
    private String province;

    // Constructor
    public Location(String address, String city, String province) {
        this.address = address;
        this.city = city;
        this.province = province;
    }

    // Method to get the full address as a string
    public String fullAddress() {
        return address + ", " + city + ", " + province;
    }

    // Overriding the toString method to return the full address
    @Override
    public String toString() {
        return fullAddress();
    }

    // Getter for city
    public String getCity() {
        return city;
    }

    // Setter for city
    public void setCity(String city) {
        this.city = city;
    }

    // Getter for province
    public String getProvince() {
        return province;
    }

    // Setter for province
    public void setProvince(String province) {
        this.province = province;
    }

    public static Location fromString(String addressString) throws IllegalArgumentException {
        String[] parts = addressString.split(",\s*");

        if (parts.length != 3) {
            throw new IllegalArgumentException("Invalid address format. Expected: 'address, city, province'");
        }

        String address = parts[0].trim();
        String city = parts[1].trim();
        String province = parts[2].trim();

        return new Location(address, city, province);
    }

    @Override
    public boolean equals(Object location) {
        if (location instanceof Location) {
            return this.fullAddress().equals(((Location) location).fullAddress());
        }
        return false;
    }
}
